.circleci/create_circleci_config.py

```diff
@@ -536,6 +542,19 @@ def create_circleci_config(folder=None):
         folder = os.getcwd()
     # Used in CircleCIJob.to_dict() to expand the test list (for using parallelism)
     os.environ["test_preparation_dir"] = folder
+
+    test_file = os.path.join(folder, "filtered_test_list.txt")
+    if os.path.exists(test_file):
+        with open(test_file) as f:
+            test_list = f.read()
+    else:
+        test_list = []
+
+    use_cache_on_main = False
+    # `setup.py` is not modified
+    if test_list != "tests":
+        use_cache_on_main = True
```

Comment:
Use the cache from `main` if `setup.py` is not modified.

Two advantages:
  
  - we don't need anymore to keep 2 cache (as currently): one for `main` and one for PRs/branches
  - we